accounting/create_db.py

```diff
@@ -20,42 +20,45 @@
-        """
-            CREATE TABLE IF NOT EXISTS association_table (
-                creation_time bigint(20)            NOT NULL,
-                mod_time      bigint(20)  DEFAULT 0 NOT NULL,
-                deleted       tinyint(4)  DEFAULT 0 NOT NULL,
-                user_name     tinytext              NOT NULL,
-                admin_level   smallint(6) DEFAULT 1 NOT NULL,
-                account       tinytext              NOT NULL,
-                shares        int(11)     DEFAULT 1 NOT NULL,
-                max_jobs      int(11)               NOT NULL,
-                max_wall_pj   int(11)               NOT NULL,
-                PRIMARY KEY   (user_name, account)
-        );"""
-    )
-    logging.info("Created association_table successfully")
-
-    # Bank Table
-    # bank_id gets auto-incremented with every new entry
-    logging.info("Creating bank_table in DB...")
-    conn.execute(
-        """
-            CREATE TABLE IF NOT EXISTS bank_table (
-                bank_id     integer PRIMARY KEY,
-                bank        text    NOT NULL,
-                parent_bank text,
-                shares      int     NOT NULL
-        );"""
-    )
-    logging.info("Created bank_table successfully")
-
-    conn.close()
+    try:
+        # open connection to database
+        logging.info("Creating Flux Accounting DB")
+        conn = sqlite3.connect("file:" + filepath + "?mode:rwc", uri=True)
+        logging.info("Created Flux Accounting DB sucessfully")
+
+        # Association Table
+        logging.info("Creating association_table in DB...")
+        conn.execute(
+            """
+                CREATE TABLE IF NOT EXISTS association_table (
+                    creation_time bigint(20)            NOT NULL,
+                    mod_time      bigint(20)  DEFAULT 0 NOT NULL,
+                    deleted       tinyint(4)  DEFAULT 0 NOT NULL,
+                    user_name     tinytext              NOT NULL,
+                    admin_level   smallint(6) DEFAULT 1 NOT NULL,
+                    account       tinytext              NOT NULL,
+                    shares        int(11)     DEFAULT 1 NOT NULL,
+                    max_jobs      int(11)               NOT NULL,
+                    max_wall_pj   int(11)               NOT NULL,
+                    PRIMARY KEY   (user_name, account)
+            );"""
+        )
+        logging.info("Created association_table successfully")
+
+        # Bank Table
+        # bank_id gets auto-incremented with every new entry
+        logging.info("Creating bank_table in DB...")
+        conn.execute(
+            """
+                CREATE TABLE IF NOT EXISTS bank_table (
+                    bank_id     integer PRIMARY KEY,
+                    bank        text    NOT NULL,
+                    parent_bank text,
+                    shares      int     NOT NULL
+            );"""
+        )
+        logging.info("Created bank_table successfully")
+
+        conn.close()
```

Comment:
if an exception happens before the `conn.close()` this won't be cleaned up properly.  perhaps should use a `finally` clause here?

Reply:
Oh yeah, I forgot about that. Just pushed a fix that adds a `finally` clause that closes the SQLite connection.